File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.go
func minOperations(s string) int {
    oneOddCnt, zeroEvenCnt := 0, 0
    flag := '1'
    
    for _, i := range s {
        if flag == i {
            oneOddCnt++
        } else {
            zeroEvenCnt++
        }
        
        if flag == '1' {
            flag = '0'
        } else {
            flag = '1'
        }
    }
    return min(zeroEvenCnt, oneOddCnt)
}

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int minOperations(string s) {
        char flag = '1';
        int oneOddCnt = 0, zeroZeroCnt = 0;
        
        for(int i=0; i<s.size(); i++) {
            if(s[i] != flag) 
                oneOddCnt++;
            else
                zeroZeroCnt++;
            
            if(flag == '1') 
                flag = '0';
             else 
                flag = '1';
        }
        
        return min(oneOddCnt, zeroZeroCnt);
    }
};